### DISCUSSION TITLE
TODO: Introduce non-release development/nightly branch

### DISCUSSION THREAD

**Discussion by freefd** [0x2142/frigate-notify#87] [General]
Dear @0x2142,

To be able to build the most recent non-released code in an automated way, a fixed branch/tag name is highly required, and I'd like to suggest the term `nightly` or `dev`.

For example, in Containerfile, it is used as follows:
```Dockerfile
...

ARG FRIGATE_NOTIFY_REPOSITORY_URL FRIGATE_NOTIFY_REPOSITORY_TAG
ENV FRIGATE_NOTIFY_REPOSITORY_URL ${FRIGATE_NOTIFY_REPOSITORY_URL:-}
ENV FRIGATE_NOTIFY_REPOSITORY_TAG ${FRIGATE_NOTIFY_REPOSITORY_TAG:-main}

WORKDIR /tmp

RUN apk add --no-cache git upx tzdata ca-certificates \
    && git clone $FRIGATE_NOTIFY_REPOSITORY_URL src \
    && cd src \
    && if [ "$FRIGATE_NOTIFY_REPOSITORY_TAG" != "main" ]; \
       then git checkout "$FRIGATE_NOTIFY_REPOSITORY_TAG"; fi \
    && go mod download \
    && go build -o frigate-notify . \
    && upx --lzma --best -k -q frigate-notify

...
```
with passing the environment variables during build:
```ini
FRIGATE_NOTIFY_REPOSITORY_URL=https://github.com/0x2142/frigate-notify
FRIGATE_NOTIFY_REPOSITORY_TAG=dev
```

produces:
```log
...

Cloning into 'src'...
branch 'dev' set up to track 'origin/dev'.
Switched to a new branch 'dev'

...
```

**Comment by 0x2142** [comment]
I set up a `dev` branch & merged my work in progress. New container builds [here](https://github.com/0x2142/frigate-notify/pkgs/container/frigate-notify/228121359?tag=dev) if you would like to test anything in particular

**Reply by freefd** [reply]
Thanks a lot!
But I only use vanilla code builds from the scratch in my network :man_shrugging: 

I think you can mention the `dev` container tag in the docs as an unstable and recent version of the code, not yet stabilized and provided as is for those in love with cutting edge tech.

**Reply by freefd** [reply]
I've just built the code from `dev` branch, looks perfect, let's close this thread as implemented :champagne:
